logger_test.go

```diff
@@ -2,11 +2,91 @@ package micrologger
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"testing"
+
+	"github.com/giantswarm/micrologger/loggercontext"
 )
 
-func TestWith(t *testing.T) {
+func Test_Logger_LogWithCtx(t *testing.T) {
```

Comment:
Just something I wondered again just now: Why are using underscores in tests? I used them myself but can't remember a reasoning other than "we do it".

Reply:
It is a structural separation to have something like blocks describing which component of a project the test is about.